### PR TITLE
Updated project documentation

### PR DESCRIPTION
**Pull request by jgebal**

### CODE REVIEW

File: docs/userguide/annotations.md [hunk 1]
```diff
@@ -160,3 +158,51 @@ A `%suitepath` can be provided in tree ways:
...
+# Using automatic rollbacks in tests
+
+By default, every test is isolated using savepoint.
+This solution is suitable for use-cases, where the code that is getting tested as well as the unit tests themselves do not use transaction control (commit/rollback) or DDL commands.
+In general, your unit tests should not use transaction control as long as the code you are testing is not using it too.
+Keeping the transactions uncommitted allows your changes to be isolated and the execution of tests is not impacting others that might be using a shared development database.
+
+If you are in situation, where the code you are testing, is using transaction control (common case with ETL code), then your tests probably should not use the default automatic transaction control.
+In that case use the annotation `-- %rollback(manual)` on the suite level to disable automatic transaction control for entire suite.
+If you are using nested suites, you need to make sure that thr entire suite all the way to the root is using manual transaction control.
```
Comment (Pazus) [inline docs/userguide/annotations.md]: typo

File: docs/userguide/annotations.md [hunk 2]
```diff
@@ -160,3 +158,51 @@ A `%suitepath` can be provided in tree ways:
...
+
+In some cases it is needed to perform DDL as part of setup or cleanup for the tests. 
+It is recommended to move such DDL statements to a procedure with `pragma autonomous_transaction` to eliminate implicit commit of the main session.
```
Comment (Pazus) [inline docs/userguide/annotations.md]: Also if no transactional setup/cleanup and test procedure validates some ETL-like functionality, test procedure should be defined with `pragma autonomous_transaction`.

Comment (jgebal) [inline docs/userguide/annotations.md]: Done

File: docs/userguide/annotations.md (hunk 1 above)
Comment (jgebal) [inline docs/userguide/annotations.md]: Fixed

**Pull request merged**